Draft.md

```diff
@@ -347,6 +392,8 @@ Schließlich ist es für Entwickler oft interessant die Software selbst mitzuges
 - Website: URL unter der weitere Informationen abgerufen werden können, z.B. "https://qgis.org"
 - Lizenzierung: Unter welcher Softwarelizenz wurde die Software veröffentlicht, z.B. "Open Source mit GNU General Public License (GPL)"
 
+> [MT] Ein- und Ausgabeformate fände ich noch ganz praktisch
```

Comment:
Sollen wir hier "gute" Formate beschreiben die wir unterstützen oder eine allgemeine Beschreibung von guten Eigenschaften von Formaten erstellen?

Reply:
Ich glaube es geht nur darum, die Ein- und Ausgabeformate einer Software in diese Übersichtsliste aufzuführen.

Ich verstehe woher du kommst, @bellerophons-pegasus: Idealerweise wäre das dann wie bei einer statisch getypten Programmiersprache, wo man am Interface sofort erkennt, ob die Software/Funktion zu seiner Anwendung passt :smile:. 

Aber ich fürchte, diese Information wäre in der Realität oft ziemlich bedeutungslos. QGIS unterstützt dutzende Formate für verschiedene Zwecke. Ein R Paket, dass tabellierte Daten oder Arrays modifiziert, schluckt theoretisch buchstäblich alles, was du in R einladen kannst. Interessant wäre das nur für wirklich spezielle Software, wie etwa im 3D Bereich. Aber ob es sich dafür lohnt, das hier einzufügen? Ich bin nicht sicher.

Reply:
Dann könnte man höchstens zwischen offenen und proprietären Formaten differenzieren würde ich sagen und darüber ob diese Formate zu einem typischen Anwendungsfall der Software passen, vll derjenige an dem man die Review auch ein wenig aufhängt - und wie gut sich dann diese Formate für diese Anwedungsfälle eignen. GeoJSON mit festem Geokoordinatensystem eignet sich zum Beispiel nicht für alle GIS Anwendungen

Reply:
Ich kann mir nicht so gut vorstellen, wie das in einer Tabelle kurz dargestellt werden könnte. Könntest du das QGIS-Beispiel hier schnell ausformulieren?

Reply:
Wenn ich das für QGIS beschreiben würde, dann würde ich das Review anhand eines typischen Anwendungsfalls aufziehen, also zum Beispiel: 
1. Anwendungsfall Vektordatenmanipulation zur Darstellung im Web:
Ich habe ein Shapefile und will das reprojizieren und dann auf einer Webseite anzeigen.
Dann bietet mir QGIS:
- Mindestens ein offenes gängiges Exportformat dafür (GeoJSON) (check!)
- Die Möglichkeit des Imports von proprietären Formaten (Shapefile) (check!)
- Die Möglichkeit auch der Bearbeitung von gängigen offenen Formaten (GeoJSON, GML, KML) (check!)
2. Beispiel: 
Vektordatenmanipulation in verschiedenen CRS mit Surfaces als Geometrien:
- QGIS bietet mir den Import in proprietären und offenen Formaten (check!)
- QGIS bietet mir nicht den Export in GML als offenes Format (nur lesend) (not check!)
- GeoJSON und KML eignen sich nicht als Export weil sie nur im Weltkoordinatensystem definiert sind und keine Surfaces unterstützen
- Mir bleiben nur noch proprietäre Formate (not check!)
-> Ich hoffe mal ich habe das so korrekt wiedergegeben
Das heisst ich kann QGIS nicht dafür verwenden Daten in GML zu speichern die GML-spezifisch sind (Surfaces z.B.)

Was ich damit sagen will ist:
Wenn ich so Aussagen treffen will dann muss ich als Reviewer sagen was ich als typische Anwendungsfälle für die Software empfinde und was dafür gängige Datenformate sind und dann eben differenzieren.
Und das müsste der Reviewer dann nach Kenntnisstand bestenfalls für die Archäologie bewerten.

Reply:
Wir reden aneinander vorbei, Timo. Was du beschreibst ist, was sich in den Stichpunkten unter `#### Dateningest, Interoperabilität und Schnittstellen` abbildet. Und das ist alles gut und richtig und wichtig. In dieser Diskussion geht es aber (meiner Wahrnehmung nach) um folgende 

```
**Software-Eckdatentabelle**

- Name: Der Name der Software, z.B. "QGIS"
- Kurzbeschreibung: Was die Software leistet, z.B. "Umfangreiches graphisches Werkzeug zur Raumdatenverarbeitung"
- Rezensierte Version: Die Softwareversion, die für das Review genutzt wurde, z.B. "3.10.10 LTR"
- Plattform: Betriebssysteme, auf denen die Software genutzt werden kann, z.B. "Windows, macOS, Linux, BSD, Android"
- Website: URL unter der weitere Informationen abgerufen werden können, z.B. "https://qgis.org"
- Lizenzierung: Unter welcher Softwarelizenz wurde die Software veröffentlicht, z.B. "Open Source mit GNU General Public License (GPL)"
```

Martina fragt, ob es hier einen Stichpunkt `Ein- und Ausgabeformate` geben sollte. Für QGIS könnte man sich das vllt. so vorstellen: 

```
Ein- und Ausgabeformate: SpatiaLite, PostGIS, MSSQL, DB2, WMD/WMTS, Vector Tiles, XYZ Tiles, WCS, WFS, OWS, ArcGIS Map Service, ArcGIS Feature Service, GeoNode, ESRI Shapefile, Geopackage, GeoTIFF, CSV, TXT, PDF, ... 
```

Ganz daneben ist es nicht -- vor allem da die meiste Software ja mit weniger Formaten auskommt. Aber was würde ich für ein R Paket schreiben, das einfach Matrizenoperationen ausführt? Da spielen Datenformate keine Rolle :monocle_face:

Reply:
Vielleicht würde man den Punkt dann einfach weglassen?

Reply:
Ja, ich meinte tatsächlich einfach eine Liste von Dateiformaten. 
Wenn Dateiformate tatsächlich keine Rolle spielen (wobie, wenn in R Daten eingelesen werden, wird vermutlich auch eine CSV mit entsprechenden Trennzeichen benötigt), muss man auch nichts in die Tabelle eintragen.

Reply:
Ich ahne, dass das zu Verwirrung führen wird. Aber in Ordnung. In unsere Beispiel/Meta-Tabelle würde man dann schreiben: 

Ein- und Ausgabeformate: Welche Dateiformate die Software verarbeiten kann, z.B. " Geodatenbanken (SpatiaLite, PostGIS, MSSQL, ...), Web-Geodaten Dienste (WMD/WMTS, Vector Tiles, XYZ Tiles, WFS, ...), Geo-Vektordatenformate (ESRI Shapefile, Geopackage, ...), Geo-Rasterdatenformate (GeoTIFF, ...), Tabellendaten (CSV, TXT, ...) und weitere Datentypen."